### PR TITLE
Don't allow names to be edited

### PR DESCRIPTION
**Pull request by timja-kainos**
Fix bugs around loading of contactPerson and dateOfBirth

### CODE REVIEW

File: src/main/features/response/views/your-details.njk [hunk 1]
```diff
@@ -16,30 +16,30 @@
...
             ]
           %}
           {% set addressSectionHeading = 'Your address' %}
         {% elif form.model.type == 'soleTrader' %}
           {%
             set fields = [
-              { label: 'Your full name', name: 'name' },
-              { label: 'Business name (optional)', name: 'businessName' }
+              { readOnly: true, label: 'Your full name', value: claim.claimData.defendant.name },
+              { readOnly: true, label: 'Business name (optional)', value: claim.claimData.defendant.businessName }
```
Comment (geek-soft) [inline src/main/features/response/views/your-details.njk]: I wonder if this still make sense to say `(optional)` for read only fields. For me that is the hint for someone who is filling the form. If value not provided then that heading won't be even displayed.

File: src/main/app/forms/models/companyDetails.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export class CompanyDetails extends PartyDetails {
 
   @IsDefined({ message: ValidationErrors.CONTACT_PERSON_REQUIRED, groups: ['claimant'] })
   @IsNotBlank({ message: ValidationErrors.CONTACT_PERSON_REQUIRED, groups: ['claimant'] })
-  @MaxLength(35, { message: ValidationErrors.CONTACT_PERSON_NAME_TOO_LONG, groups: ['claimant'] })
+  @MaxLength(35, { message: ValidationErrors.CONTACT_PERSON_NAME_TOO_LONG, groups: ['claimant', 'response'] })
```
Comment (geek-soft) [inline src/main/app/forms/models/companyDetails.ts]: That makes sense, something we forgot earlier 👍

File: src/main/features/response/routes/your-details.ts [hunk 3]
```diff
@@ -44,7 +47,18 @@ export default express.Router()
 
     const partyDetails: PartyDetails = plainToClass(PartyDetails, user.claim.claimData.defendant)
     if (user.responseDraft.defendantDetails.partyDetails) {
-      partyDetails.name = user.responseDraft.defendantDetails.partyDetails.name
+      switch (user.responseDraft.defendantDetails.partyDetails.type) {
```
Comment (geek-soft) [inline src/main/features/response/routes/your-details.ts]: Looks like fix for something we missed the other day.

I wonder if we could make it even better by pointing the right type to `plainToClass(<here>, user.claim.claimData.defendant)`.

Then we could extract type from `user.claim.claimData.defendant.type`to find constructor for given party type.

To summarise something like `plainToClass(PartyDetailsFactory.constructorFor(user.claim.claimData.defendant.type), user.claim.claimData.defendant)`

File: src/main/features/response/views/your-details.njk (hunk 1 above)
Comment (timja-kainos) [reply src/main/features/response/views/your-details.njk]: will doublecheck

Comment (timja-kainos) [reply src/main/features/response/views/your-details.njk]: confirmed, remove optional with Art

File: src/main/features/response/routes/your-details.ts (hunk 3 above)
Comment (timja-kainos) [reply src/main/features/response/routes/your-details.ts]: will give it a go, but do need to get this in soon for release next week +1

Comment (geek-soft) [reply src/main/features/response/routes/your-details.ts]: Theoretically it should work. I thought that when pass any constructor for class extending from `TheirDetails` it should nicely deserialise it but I never tested subclasses with this library. Happy to leave it as is for now and rework it when we start using the library more extensively.

Comment (timja-kainos) [reply src/main/features/response/routes/your-details.ts]: subclassing does work

**Pull request merged**